src/controllers/resize.ts

```diff
@@ -0,0 +1,59 @@
+import { Context } from 'koa'
+import { Controller, Ctx, Get, Params, Query } from 'amala'
+import { IPFS_TIMEOUT } from '../constants'
+import { Readable } from 'stream'
+import { badRequest, internal, notFound } from '@hapi/boom'
+import { getCachedResize, setCachedResize } from '../helpers/resizedImageCache'
+import { getContentType } from '../helpers/contentTypes'
+import CIDParam from '../validators/CIDParam'
+import ipfs, { getDataFromIPFS } from '../helpers/ipfs'
+import resizeImage from '../helpers/resizeImage'
+
+@Controller('/resize')
+export default class ResizeController {
+  @Get('/image/:cid')
+  async image(
+    @Ctx() ctx: Context,
+    @Params() params: CIDParam,
+    @Query('width') width?: number,
```

Comment:
Do we check types of `Query`? Make sure we can't pass `width=abc` or something like that

Reply:
Fixed in [391eeec](https://github.com/BigWhaleLabs/obss-ipfs-uploader/pull/12/commits/391eeec272189aa9844464d0a940c00b29fced12). I thought amala was validating this type automatically, but apparently not.